### PR TITLE
Alteration Draft filings, Shared FeeSummary, Actions Flow updates

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#5167

*Description of changes:*

* Imports and implements Shared Fee Summary component.

* Refactors the actions and flow for Alterations.

* Implements functionality to CREATE/BUILD/PARSE Alteration filings in filing mixin.

* Save/Resume Later DRAFT alterations.

* Cancel Alteration Filing.

* Proceed to summary view of Alterations (Summary view NOT complete, but the flag for it is in place).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

File: src/interfaces/filing-interfaces/alterationFiling-interface.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
-import { AlterationIF } from '@/interfaces'
+import { AlterationIF, BusinessSnapshotIF } from '@/interfaces'
```
Comment (severinbeauvais) [inline src/interfaces/filing-interfaces/alterationFiling-interface.ts:1]: `BusinessSnapshotIF` is for future use? (Is not used here atm.)

File: src/interfaces/alteration-interfaces/alteration-interface.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import {
   BusinessContactIF,
-  BusinessInformationIF,
+  BusinessInformationIF, NameRequestApplicantIF,
```
Comment (severinbeauvais) [inline src/interfaces/alteration-interfaces/alteration-interface.ts:3]: Looks like `NameRequestApplicantIF` isn't used here.

File: src/mixins/legal-api-mixin.ts [hunk 3]
```diff
@@ -82,6 +82,33 @@ export default class LegalApiMixin extends Mixins(FilingTemplateMixin) {
     }
 
     return axios.put(url, { filing }).then(response => {
+      console.log(response)
```
Comment (severinbeauvais) [inline src/mixins/legal-api-mixin.ts:85]: Remove debug code.

File: src/store/state/state-model.ts [hunk 4]
```diff
@@ -137,7 +138,9 @@ export const stateModel: StateModelIF = {
         identifier: ''
       },
       nameRequest: {
-        legalType: ''
+        legalType: null,
+        nrNumber: '',
+        legalName: ''
```
Comment (severinbeauvais) [inline src/store/state/state-model.ts:143]: You might consider an "empty" nameRequest object that you can assign here.

Eg, https://github.com/bcgov/business-filings-ui/blob/091148a18ab9e62497eb2b12cbce256af609536f/src/interfaces/address-interfaces.ts#L26

File: src/interfaces/alteration-interfaces/alteration-interface.ts (hunk 2 above)
Comment (cameron-freshworks) [reply src/interfaces/alteration-interfaces/alteration-interface.ts:3]: thanks! Will remove if obsolete

Comment (cameron-freshworks) [reply src/interfaces/alteration-interfaces/alteration-interface.ts:3]: removed in next commit!

File: src/interfaces/filing-interfaces/alterationFiling-interface.ts (hunk 1 above)
Comment (cameron-freshworks) [reply src/interfaces/filing-interfaces/alterationFiling-interface.ts:1]: oh thanks good catch! 

Initially i was planning to store the original snapshot within the DRAFT alteration filings, but after our discussion we determined we didn't need too and could GET the current information.

Comment (cameron-freshworks) [reply src/interfaces/filing-interfaces/alterationFiling-interface.ts:1]: Removed in next commit!

File: src/mixins/legal-api-mixin.ts (hunk 3 above)
Comment (cameron-freshworks) [reply src/mixins/legal-api-mixin.ts:85]: Thank you! Removed in next commit

File: src/store/state/state-model.ts (hunk 4 above)
Comment (cameron-freshworks) [reply src/store/state/state-model.ts:143]: I like it! Implemented in next commit

File: src/store/state/state-model.ts [hunk 5]
```diff
@@ -137,11 +137,7 @@ export const stateModel: StateModelIF = {
         legalType: '',
         identifier: ''
       },
-      nameRequest: {
-        legalType: null,
-        nrNumber: '',
-        legalName: ''
-      },
+      nameRequest: EmptyNameRequest,
```
Comment (severinbeauvais) [inline src/store/state/state-model.ts:140]: You should probably do `{ ...EmptyNameRequest }` so you're not pointing to the original object.

Comment (cameron-freshworks) [reply src/store/state/state-model.ts:140]: dooh, will do!

Comment (cameron-freshworks) [reply src/store/state/state-model.ts:140]: good ole javascript. Updated in next commit!

**Pull request merged**